modules/apps/portal-remote/portal-remote-jaxrs-whiteboard/src/main/java/com/liferay/portal/remote/jaxrs/whiteboard/lifecycle/JAXRSLifecycleUtil.java

```diff
@@ -34,20 +32,15 @@ Object.class, new Object(),
 					"liferay.jaxrs.whiteboard.ready", true)));
 	}
 
-	@Activate
-	protected void activate(BundleContext bundleContext) {
-		_bundleContext = bundleContext;
-	}
+	private static final BundleContext _bundleContext;
+	private static boolean _jaxrsReady;
+	private static final DCLSingleton<ServiceRegistration<?>>
+		_serviceRegistrationDCLSingleton = new DCLSingleton<>();
 
-	@Deactivate
-	protected void deactivate() {
-		_serviceRegistrationDCLSingleton.destroy(
-			ServiceRegistration::unregister);
-	}
+	static {
+		Bundle bundle = FrameworkUtil.getBundle(JAXRSLifecycleUtil.class);
```

Comment:
@GabrielHenriquedev You do not need the static block here. Just get the bundleContext into the `ensureReady` method. Thx